OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -395,81 +395,9 @@ computeActuation(const SimTK::State& s) const
-        return;
-    }
-
-    // Elastic tendon initialization routine.
-
-    // Initialize activation as specified by the user.
-    double clampedActivation =
-        getActivationModel().clampActivation(getActivation(s));
-    setActivation(s,clampedActivation);
-
-    // Initialize the multibody system to the initial state vector.
-    setFiberLength(s, getOptimalFiberLength());
-    _model->getMultibodySystem().realize(s, SimTK::Stage::Velocity);
-
-    // Compute an initial muscle state that develops the desired force and
-    // shares the muscle stretch between the muscle fiber and the tendon
-    // according to their relative stiffnesses.
-
-    // tol is the desired tolerance in Newtons.
-    double tol = 1e-8*getMaxIsometricForce();
-    if(tol < SimTK::SignificantReal*10) {
-        tol = SimTK::SignificantReal*10;
-    }
-    int maxIter = 200;
-    double pathLength = getLength(s);
-    double pathLengtheningSpeed = getLengtheningSpeed(s);
-
-    try {
-        std::pair<StatusFromEstimateMuscleFiberState,
-                  ValuesFromEstimateMuscleFiberState> result =
-            estimateMuscleFiberState(clampedActivation, pathLength,
-                                     pathLengtheningSpeed, tol, maxIter);
-
-        switch(result.first) {
-
-        case StatusFromEstimateMuscleFiberState::Success_Converged:
-            setActuation(s, result.second["tendon_force"]);
-            setFiberLength(s, result.second["fiber_length"]);
-            break;
-
-        case StatusFromEstimateMuscleFiberState::Warning_FiberAtLowerBound:
-            printf("\n\nMillard2012EquilibriumMuscle initialization:"
-                   " %s is at its minimum fiber length of %f\n",
-                   getName().c_str(), result.second["fiber_length"]);
-            setActuation(s, result.second["tendon_force"]);
-            setFiberLength(s, result.second["fiber_length"]);
-            break;
-
-        case StatusFromEstimateMuscleFiberState::Failure_MaxIterationsReached:
-            // Report internal variables and throw exception.
-            std::ostringstream ss;
-            ss << "\n  Solution error " << abs(result.second["solution_error"])
-               << " exceeds tolerance of " << tol << "\n"
-               << "  Newton iterations reached limit of " << maxIter << "\n"
-               << "  Clamped activation is " << clampedActivation << "\n"
-               << "  Fiber length is " << result.second["fiber_length"] << "\n";
-            OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, ss.str());
-            break;
-        }
-
-    } catch (const std::exception& x) {
-        OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate,
-            "Internal exception encountered.\n" + std::string{x.what()});
-    }
-}
 
 void Millard2012EquilibriumMuscle::
-computeFiberEquilibriumAtZeroVelocity(SimTK::State& s) const
+computeFiberEquilibrium(SimTK::State& s, bool useZeroVelocity) const
```

Comment:
Nitpicky: consider flipping the meaning of the new parameter. Because `0` evaluates to false, it may make more sense for `false` to mean "zero velocity". The name of the parameter could instead be "solveForVelocity."